### PR TITLE
fix: provide better error message when unknown feature flags are present

### PR DESCRIPTION
**Pull request by attilah**
*Description of changes:*

When a newer version is used to push a project, and on another machine or CI/CD an older version executes any Amplify CLI command, then the CLI shows an error message about extra members in a JSON file.

This PR changes the validation and provides a meaningful error message to customers. An example:
```sh
Invalid feature flag configuration

The following feature flags are defined in configuration are unknown by the currently running version of Amplify CLI:
  - baz
  - graphqltransformer.foo
  - graphqltransformer.bar

The following feature flags have validation errors:
  - graphqltransformer.useexperimentalpipelinedtransformer: should be boolean

The project was pushed with a newer version of the Amplify CLI that is being used now. Update to the newer version that supports the feature flags shown as unknown above.

Learn more about feature flags at: https://docs.amplify.aws/cli/reference/feature-flags
``` 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by yuth — APPROVED** [review]
👍 💯 :shipit:

File: packages/amplify-cli/src/index.ts [hunk 1]
```diff
@@ -116,14 +119,61 @@ export async function run() {
...
+    if (error.name === 'JSONValidationError') {
+      const jsonError = <JSONValidationError>error;
+      let printSummary = false;
+
+      print.error(error.message);
+
+      if (jsonError.unknownFlags?.length > 0) {
+        print.error('');
+        print.error(
+          `The following feature flags are defined in configuration are unknown by the currently running version of Amplify CLI:`,
```
Comment (renebrandel) [inline packages/amplify-cli/src/index.ts:135]: These feature flags are defined in the amplify/cli.json configuration file and are unknown to the currently running Amplify CLI:

File: packages/amplify-cli/src/index.ts [hunk 2]
```diff
@@ -116,14 +119,61 @@ export async function run() {
...
+          print.error(`  - ${otherError}`);
+        }
+
+        printSummary = true;
+      }
+
+      if (printSummary) {
+        print.error('');
+        print.error(
+          `The project was pushed with a newer version of the Amplify CLI that is being used now. Update to the newer version that supports the feature flags shown as unknown above.`,
```
Comment (renebrandel) [inline packages/amplify-cli/src/index.ts:159]: This issue likely happens when the project has been pushed with a newer version of Amplify CLI. Try updating to a newer version of Amplify CLI by running "amplify upgrade"

File: packages/amplify-cli/src/index.ts [hunk 3]
```diff
@@ -116,14 +119,61 @@ export async function run() {
...
+        );
+
+        if (isCI) {
+          print.error('');
+          print.error(`Make sure that the CI/CD pipeline is not using an older, pinned down version of the Amplify CLI.`);
```
Comment (renebrandel) [inline packages/amplify-cli/src/index.ts:164]: Ensure that the CI/CD pipeline is not using an older or pinned down version of Amplify CLI.

File: packages/amplify-cli/src/index.ts [hunk 4]
```diff
@@ -116,14 +119,61 @@ export async function run() {
...
+        }
+
+        print.error('');
+        print.error(`Learn more about feature flags at: https://docs.amplify.aws/cli/reference/feature-flags`);
```
Comment (renebrandel) [inline packages/amplify-cli/src/index.ts:168]: Learn more about feature flags: <link>

**Pull request merged**